data-pipeline/src/stats_exporter.rs

```diff
@@ -0,0 +1,362 @@
+// Copyright 2024-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{
+    sync::{
+        atomic::{AtomicU64, Ordering},
+        Arc, Mutex,
+    },
+    time,
+};
+
+use datadog_trace_protobuf::pb;
+use ddcommon::{connector, tag::Tag, Endpoint};
+use hyper;
+use tokio::select;
+use tokio_util::sync::CancellationToken;
+
+use crate::span_concentrator::SpanConcentrator;
+
+const STATS_ENDPOINT_PATH: &str = "/v0.6/stats";
+
+/// Metadata required in a ClientStatsPayload
+#[derive(Debug, Default, Clone)]
+pub struct LibraryMetadata {
+    pub hostname: String,
+    pub env: String,
+    pub version: String,
+    pub lang: String,
+    pub tracer_version: String,
+    pub runtime_id: String,
+    pub service: String,
+    pub container_id: String,
+    pub git_commit_sha: String,
+    /// Should be left empty by client, except for some specific environment
+    pub tags: Vec<Tag>,
+}
+
+/// An exporter sending flushing a SpanConcentrator based on a timer and sending stats to the agent
+#[derive(Debug)]
+pub struct StatsExporter {
+    flush_interval: time::Duration,
+    concentrator: Arc<Mutex<SpanConcentrator>>,
+    endpoint: Endpoint,
+    meta: LibraryMetadata,
+    sequence_id: AtomicU64,
+    client: ddcommon::HttpClient,
+    cancellation_token: CancellationToken,
+}
+
+impl StatsExporter {
+    /// Return a new StatsExporter
+    ///
+    /// - `meta` is used when sending the ClientStatsPayload to the agent
+    /// - `cfg` is the configuration for the stats exporter
+    ///
+    /// Returns a result to have the same signature as the blocking implementation.
```

Comment:
I don't understand this comment. It returns a `StatsExporter` and not a `result`, or am I missing something?

Reply:
My bad, this doc is outdated since I removed the blocking call. I updated it.